### PR TITLE
VizTooltips: Remove old tooltips and annotations

### LINKED ISSUE #83632: Remove old tooltips and annotations
before we can remove old tooltips, we need to deploy `newVizTooltips` in HG, and validate parity with old tooltips.

this is a sub-epic of https://github.com/grafana/grafana/issues/68238

we have started rollout of this feature toggle internally to `dev` and `staging` for dogfooding and feedback: https://github.com/grafana/deployment_tools/pull/124451

**DRY**

- [x] Extract `EventBus` publisher and subscriber to plugin instead of duplicating code in each panel's `prepConfig()`
  - [x] Verify bi-directional cursor/hover sync between Table and uPlot-based panels: https://github.com/grafana/grafana/pull/78392
- [x] Rework Heatmap tooltip code to share dataLink logic with other tooltips: https://github.com/grafana/grafana/pull/71016
- [x] Unify and move cursor sync setup into `TooltipPlugin2` from various `prepConfig()`: https://github.com/grafana/grafana/pull/84420

**Verify datalinks work correctly in all panels**

- [ ] StatusHistory, StateTimeline
  - [ ] Re-check https://github.com/grafana/grafana/pull/50226, https://github.com/grafana/support-escalations/issues/932
- [ ] TimeSeries
- [ ] Candlestick
- [ ] Trend
- [ ] BarChart
- [ ] XYChart
- [ ] Heatmap? (only non-numeric heatmap-rows case?)

**Verify exemplars tooltip works correctly**

 - [ ] TimeSeries
 - [ ] Heatmap

**Verify other plugins play nice**

- [ ] KeyboardPlugin (all x=time)
  - [ ] TimeSeries
  - [ ] StateTimeline
  - [ ] StatusHistory
  - [ ] Candlestick
  - [ ] Heatmap
- [ ] OutsideRangePlugin (all x=time)
  - [ ] Fix to check `null` in values, not in timestamps since https://github.com/grafana/grafana/pull/57335
  - [ ] TimeSeries
  - [ ] StateTimeline
  - [ ] StatusHistory
  - [ ] Candlestick
  - [ ] Heatmap
- [ ] ThresholdControlsPlugin
  - [ ] TimeSeries
- [ ] ExemplarsPlugin
  - [ ] TimeSeries

**Remove old tooltip code from panels**

- [ ] StateTimeline
- [ ] StatusHistory
- [ ] BarChart
- [ ] Trend
- [ ] Candlestick
- [ ] TimeSeries
- [ ] Heatmap
- [ ] XYChart

**Remove now-unused components and features**

unserious-but-good reference: https://github.com/grafana/grafana/pull/83557

- [ ] addTooltipSupport
- [ ] PlotTooltipInterpolator
- [ ] AnnotationEditor + AnnotationTooltip https://github.com/grafana/grafana/tree/main/public/app/plugins/panel/timeseries/plugins/annotations
- [ ] AnnotationsPlugin: https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/timeseries/plugins/AnnotationsPlugin.tsx
- [ ] AnnotationEditorPlugin: https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/timeseries/plugins/AnnotationEditorPlugin.tsx
- [ ] ZoomPlugin: https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/uPlot/plugins/ZoomPlugin.tsx
- [ ] TooltipPlugin: https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin.tsx
- [ ] ContextMenuPlugin: https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/timeseries/plugins/ContextMenuPlugin.tsx
- [ ] StateTimelineTooltip: https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/state-timeline/StateTimelineTooltip.tsx
- [ ] StatusHistoryTooltip: https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/status-history/StatusHistoryTooltip.tsx
- [ ] HeatmapHoverViewOld: https://github.com/grafana/grafana/blob/main/public/app/plugins/panel/heatmap/HeatmapHoverViewOld.tsx
- [ ] SeriesTable (need to remove single remaining use in PieChart)? https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/VizTooltip/SeriesTable.tsx

### PR DESCRIPTION
**Pull request by leeoniya**
Fixes https://github.com/grafana/grafana/issues/83632

the way to review this is not to look at the code :sweat_smile:. but just to make sure the new tooltips continue to work as before :+1:

### CODE REVIEW

File: packages/grafana-ui/src/components/uPlot/config/UPlotConfigBuilder.ts [hunk 1]
```diff
@@ -30,17 +30,18 @@
 type PreDataStacked = (frames: DataFrame[], stackingGroups: StackingGroup[]) => AlignedData | FacetedData;
 
 export class UPlotConfigBuilder {
+  readonly uid = Math.random().toString(36).slice(2);
```
[the remark this answers is not in the document]
Comment (leeoniya) [reply packages/grafana-ui/src/components/uPlot/config/UPlotConfigBuilder.ts:33]: this is not a security context

**Review by gillesdemey — APPROVED** [review]
Alerting changes LGTM 🚀

File: public/app/features/alerting/unified/components/rules/state-history/LogTimelineViewer.tsx [hunk 2]
```diff
@@ -20,7 +19,6 @@ const replaceVariables: InterpolateFunction = (v) => v;
 
 export const LogTimelineViewer = React.memo(({ frames, timeRange, onPointerMove = noop }: LogTimelineViewerProps) => {
   const theme = useTheme2();
-  const { setupCursorTracking } = useCursorTimelinePosition(onPointerMove);
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/rules/state-history/LogTimelineViewer.tsx:23]: Small nit: can we remove `onPointerMove` from the interface / props here? It no longer seems to be used :)

Comment (leeoniya) [reply public/app/features/alerting/unified/components/rules/state-history/LogTimelineViewer.tsx:23]: :+1:

Comment (leeoniya) [reply public/app/features/alerting/unified/components/rules/state-history/LogTimelineViewer.tsx:23]: also had to remove it from LokiStateHistory in https://github.com/grafana/grafana/pull/84420/commits/d8dbefb6dc1aeb5d12ae7529b97c425ee048e605

**Review by adela-almasan — APPROVED** [review]

**Pull request merged**